.ci/test_r_package.sh

```diff
@@ -163,11 +163,12 @@ elif [[ $R_BUILD_TYPE == "cran" ]]; then
         || (cat ${RCHK_LOG_FILE} && exit -1)
         cat ${RCHK_LOG_FILE}
 
-        # the exception below is from R itself and not LightGBM:
+        # the exceptions below are from R itself and not LightGBM:
         # https://github.com/kalibera/rchk/issues/22#issuecomment-656036156
         exit $(
             cat ${RCHK_LOG_FILE} \
             | grep -v "in function strptime_internal" \
+            | grep -v "in function RunGenCollect" \
```

Comment:
This is to address the following new error from `rchk` on R 4.2.2.

> ERROR: too many states (abstraction error?) in function RunGenCollect

<details><summary>full rchk log (click me)</summary>

```text
Library name (usually package name): lightgbm
Initialization function: R_init_lightgbm
Functions: 41
Checked call to R_registerRoutines: 1
ERROR: too many states (abstraction error?) in function strptime_internal
ERROR: too many states (abstraction error?) in function RunGenCollect
Analyzed 40027 functions, traversed 105616 states.

Rchk version: 29e18222e1cfa67ba8e63bd3b22a25f78593c554
R version: 83396/R Under development (unstable) (2022-12-02 r83396)
LLVM version: 14.0.0
```

</details>

I think it's ok to skip it, based on https://github.com/kalibera/rchk/issues/22#issuecomment-656036156, for the same reason we skip the warning about `strptime_internal`.

`RunGenCollect` is from R's internals, not LightGBM's code.